### ISSUE TITLE
The plugin resolves the compile classpath eagerly

### ISSUE THREAD

**Issue by borissmidt** [cosmicsilence/gradle-scalafix#49]
When i run my gradle project which also uses the scala-crossbuild plugin
i get the following errors becuase this happens after the evaluation of your plugin:
```
Failed to notify build listener.
> Cannot change resolution strategy of dependency configuration ':project-api:crossBuildV211CompileClasspath' after it has been resolved
```

I notice that this is caused by the check for the scala version.
`ScalaSourceSet.isScalaSourceSet(project, sourceSet)`

It would be best to delay this configuration to the moment the task runs should be configured just before the task runs using lazy evaluation. 

I already made the getFullClasspath lazy by doing
```
    Provider<Set<File>> getFullClasspath() {
        return classesDirs.elements
            .flatMap{
                files -> jarDependencies.elements.map{
                    otherFiles -> files + otherFiles
                }
            }.map{it.collect{it.asFile}
        }
    }
```

an easy work around is to provide a property that can be set with the scala version on the plugin which will avoid the resolution of the scala version.

**Comment by marcelocenerine** [comment]
Hi @borissmidt . Thanks for reporting this issue (same root cause as in #19). Indeed the call to `ScalaSourceSet.getScalaVersion` triggers dependency resolution, which is not ideal and can cause problems in some projects :(

**Comment by marcelocenerine** [comment]
Passing the Scala version via a new property could be a good compromise. Another option (which I haven't tested yet) would be to copy the scala dependency into a detached configuration and resolve that configuration only (which shouldn't conflict with anything else).

`ScalaSourceSet.getFullClasspath` is easier to go about as that value is only needed when the task runs, so the plugin could pass it via a provider like you suggested

**Comment by aishfenton** [comment]
Seeing a related problem. We have a large repo with dozens of sub-modules. The plugin works, but it's causing the configure step to do a full dependency resolution for every run (even for things like `./gradle projects`). Since ours is a large repo, that a significant slow down (40s +).

**Comment by marcelocenerine** [comment]
@aishfenton, it's related indeed. Even though the scalafix tasks are lazily configured (meaning they are not configured at all unless you run them), there is a `scalaSourceSet.scalaVersion` call being performed prior to registering those tasks. That is triggering the dependency resolution of all source sets 😞

**Comment by marcelocenerine** [comment]
closed via #52

**Comment by marcelocenerine** [comment]
@borissmidt / @aishfenton , please give `0.1.11` a try and let me know how if you hit any new issue
